File: DevExpress-Demo/CS/SchedulingDemo.Wpf/ViewModels/Highlighted/FilteringDemoViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.ObjectModel;

namespace SchedulingDemo.ViewModels {
    public class FilteringDemoViewModel {
        protected FilteringDemoViewModel() {
            if(ViewModelBase.IsInDesignMode)
                return;
            foreach(var x in ReceptionDeskData.Doctors)
                x.IsVisible = false;
            foreach(var x in ReceptionDeskData.DepartmentPrimaryCare.Doctors)
                x.IsVisible = true;
            foreach(var x in ReceptionDeskData.DepartmentOphthalmology.Doctors)
                x.IsVisible = true;
        }

        public DateTime StartDate { get { return ReceptionDeskData.BaseDate; } }
        public virtual ObservableCollection<Doctor> Doctors { get { return ReceptionDeskData.Doctors; } }
        public virtual ObservableCollection<MedicalAppointment> MedicalAppointments { get { return ReceptionDeskData.Appointments; } }
        public virtual ObservableCollection<HospitalDepartment> HospitalDepartments { get { return ReceptionDeskData.Departments; } }
        public virtual ObservableCollection<Patient> Patients { get { return ReceptionDeskData.Patients; } }
        public virtual ObservableCollection<AppointmentLocation> Locations { get { return ReceptionDeskData.AppointmentLocations; } }
        public virtual ObservableCollection<PaymentState> PaymentStates { get { return ReceptionDeskData.PaymentStates; } }
    }
}

File: DevExpress-Demo/VB/SchedulingDemo.Wpf/ViewModels/Highlighted/FilteringDemoViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.ObjectModel

Namespace SchedulingDemo.ViewModels

    Public Class FilteringDemoViewModel

        Protected Sub New()
            If ViewModelBase.IsInDesignMode Then Return
            For Each x In ReceptionDeskData.Doctors
                x.IsVisible = False
            Next

            For Each x In DepartmentPrimaryCare.Doctors
                x.IsVisible = True
            Next

            For Each x In DepartmentOphthalmology.Doctors
                x.IsVisible = True
            Next
        End Sub

        Public ReadOnly Property StartDate As Date
            Get
                Return BaseDate
            End Get
        End Property

        Public Overridable ReadOnly Property Doctors As ObservableCollection(Of Doctor)
            Get
                Return ReceptionDeskData.Doctors
            End Get
        End Property

        Public Overridable ReadOnly Property MedicalAppointments As ObservableCollection(Of MedicalAppointment)
            Get
                Return ReceptionDeskData.Appointments
            End Get
        End Property

        Public Overridable ReadOnly Property HospitalDepartments As ObservableCollection(Of HospitalDepartment)
            Get
                Return Departments
            End Get
        End Property

        Public Overridable ReadOnly Property Patients As ObservableCollection(Of Patient)
            Get
                Return ReceptionDeskData.Patients
            End Get
        End Property

        Public Overridable ReadOnly Property Locations As ObservableCollection(Of AppointmentLocation)
            Get
                Return AppointmentLocations
            End Get
        End Property

        Public Overridable ReadOnly Property PaymentStates As ObservableCollection(Of PaymentState)
            Get
                Return ReceptionDeskData.PaymentStates
            End Get
        End Property
    End Class
End Namespace
